primitives/avail/src/header/extension/mod.rs

```diff
@@ -0,0 +1,68 @@
+use codec::{Decode, Encode};
+#[cfg(feature = "std")]
+use parity_util_mem::{MallocSizeOf, MallocSizeOfOps};
+use scale_info::TypeInfo;
+#[cfg(feature = "std")]
+use serde::{Deserialize, Serialize};
+use sp_core::{RuntimeDebug, H256};
+use sp_runtime_interface::pass_by::PassByCodec;
+
+pub mod v1;
+
+#[cfg(feature = "header-backward-compatibility-test")]
+pub mod v_test;
+
+/// Header extension data.
+#[derive(PartialEq, Eq, Clone, RuntimeDebug, TypeInfo, Encode, Decode, PassByCodec)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub enum HeaderExtension {
```

Comment:
So we just add new header constructor for a new version, and code will serialize it and use that version after deserialization? And use old version for old headers? Really cool if true :)

Reply:
If we add new function to header, we are forced to implement it to older versions and that forces backward-compatibility?